input/input_driver.h

```diff
@@ -101,11 +101,11 @@ struct retro_keybind
 
    /* Joypad key. Joypad POV (hats)
     * are embedded into this key as well. */
-   uint64_t joykey;
+   uint16_t joykey;
 
    /* Default key binding value -
     * for resetting bind to default */
-   uint64_t def_joykey;
+   uint16_t def_joykey;
```

Comment:
I'm not familiar with RetroArch, but you probably can't just change the structure in a "driver" without causing incompatibility.

I assume the reason why this field is 64 bit now, is for compatibility in the future: avoid breaking compatibility with existing software using these driver interfaces, when *more* than 16 bit are required.

If you worry about less than ~1k (less than pagesize, in this case possibly shorter than cacheline) of storage on a multi giga-byte devices, you probably shouldn't run modern software.